.mvn/gradle-enterprise-custom-user-data.groovy

```diff
@@ -96,3 +96,20 @@ if (System.env.GITHUB_ACTIONS) {
     }
 }
 
+// Check current Maven version and Maven Wrapper version  alignment
+def runtimeInfo = (org.apache.maven.rtinfo.RuntimeInformation) session.lookup("org.apache.maven.rtinfo.RuntimeInformation")
+def currentMavenVersion = runtimeInfo?.getMavenVersion()
+Properties mavenWrapperProperties = new Properties()
+File mavenWrapperPropertiesFile = new File(".mvn/wrapper/maven-wrapper.properties")
+if(mavenWrapperPropertiesFile.exists()) {
+    mavenWrapperPropertiesFile.withInputStream {
+        mavenWrapperProperties.load(it)
+    }
+    if(regexp = mavenWrapperProperties."distributionUrl" =~ /.*\/apache-maven-(.*)-bin\.zip/) {
+        def wrapperMavenVersion = regexp.group(1)
+        if (currentMavenVersion && wrapperMavenVersion && wrapperMavenVersion != currentMavenVersion) {
+            log.warn("Maven Wrapper is configured with a different version (" + wrapperMavenVersion + ") than the current one (" + currentMavenVersion + "). This will negatively impact build caching.")
+            buildScan.tag("misaligned-maven-version")
```

Comment:
would it be valuable to add a custom value with the wrapper version? I'm assuming that the runtime version is already part of the scan

Reply:
I didn't think it was necessary at first thought as the version can be obtained in the Git repo, let's add it to simplify investigation if any 👍

Reply:
Good point. I'm not sure it is needed if it is in the repo then. It was just a thought.

Reply:
I'll add it, Guillaume will decide if they want it or not